src/crimson/os/seastore/omap_manager/btree/omap_btree_node_impl.h

```diff
@@ -256,14 +403,58 @@ struct OMapLeafNode
     buffer.replay(*this);
   }
 
+  btreenode_pos_t get_node_split_pivot() const {
+    return this->get_split_pivot().get_offset();
+  }
+
   std::ostream &print_detail_l(std::ostream &out) const final;
 
-  std::pair<internal_iterator_t, internal_iterator_t>
+  std::pair<internal_const_iterator_t, internal_const_iterator_t>
   get_leaf_entries(std::string &key);
-
 };
 using OMapLeafNodeRef = OMapLeafNode::OMapLeafNodeRef;
 
+using omap_load_extent_iertr = OMapNode::base_iertr;
+template <typename T>
+requires std::is_same_v<OMapInnerNode, T> || std::is_same_v<OMapLeafNode, T>
+omap_load_extent_iertr::future<TCachedExtentRef<T>>
+omap_load_extent(
+  omap_context_t oc,
+  laddr_t laddr,
+  depth_t depth,
+  std::string begin,
+  std::string end,
+  std::optional<child_pos_t<OMapInnerNode>> chp = std::nullopt)
+{
+  LOG_PREFIX(omap_load_extent);
+  assert(end <= END_KEY);
+  auto size = std::is_same_v<OMapInnerNode, T>
+    ? OMAP_INNER_BLOCK_SIZE : get_leaf_size(oc.type);
+  return oc.tm.read_extent<T>(
+    oc.t, laddr, size,
+    [begin=std::move(begin), end=std::move(end), FNAME,
+    oc, chp=std::move(chp)](T &extent) mutable {
+      extent.init_range(std::move(begin), std::move(end));
+      if (extent.T::base_child_t::is_parent_valid()
```

Comment:
Same here, I suppose we can drop this condition and assert !is_parent_valid() for non-root, once the unnecessary call of `maybe_init` is removed above.

Reply:
related to https://github.com/ceph/ceph/pull/61936#discussion_r2041477805